### PR TITLE
SapMachine (16) #820: Bring Vitals up to date

### LINKED ISSUE #820: Bring Vitals up to date
Vitals development happens upstream in https://github.com/tstuefe/jdk/tree/sapmachine-vitals. Bring the latest version down to SapMachine.

### PR DESCRIPTION
**Pull request by RealCLanger**
Cherry-Pick Vitals update to sapmachine 16

fixes #820

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @RealCLanger, this pull request fulfills all formal requirements.

**Review by tstuefe — APPROVED** [review]

File: src/hotspot/share/vitals/vitals.cpp [hunk 1]
```diff
@@ -24,38 +25,37 @@
  */
 
 #include "precompiled.hpp"
-
+#include "jvm_io.h"
```
Comment (tstuefe) [inline src/hotspot/share/vitals/vitals.cpp:28]:
```suggestion
#include "jvm.h"
```

File: src/hotspot/share/vitals/vitals.cpp [hunk 2]
```diff
@@ -24,38 +25,37 @@
...
 
 #include "gc/shared/collectedHeap.hpp"
 #include "classfile/classLoaderDataGraph.inline.hpp"
 #include "code/codeCache.hpp"
 #include "memory/allocation.hpp"
 #include "memory/universe.hpp"
 #include "runtime/os.hpp"
+#include "runtime/mutex.hpp"
+#include "runtime/mutexLocker.hpp"
+#include "runtime/nonJavaThread.hpp"
```
Comment (tstuefe) [inline src/hotspot/share/vitals/vitals.cpp:38]: Remove that line

```suggestion
```

**Review by tstuefe — CHANGES_REQUESTED** [review]
Hi Christioph,

thanks for doing this. Ignore my earlier approval, this needs some fixes to build with 16:

```
--- a/src/hotspot/share/vitals/vitals.cpp
+++ b/src/hotspot/share/vitals/vitals.cpp
@@ -25,7 +25,7 @@
  */
 
 #include "precompiled.hpp"
-#include "jvm_io.h"
+#include "jvm.h"
 
 #include "gc/shared/collectedHeap.hpp"
 #include "classfile/classLoaderDataGraph.inline.hpp"
@@ -35,7 +35,6 @@
 #include "runtime/os.hpp"
 #include "runtime/mutex.hpp"
 #include "runtime/mutexLocker.hpp"
-#include "runtime/nonJavaThread.hpp"
 #include "runtime/thread.hpp"
 #include "services/memTracker.hpp"
 #include "services/mallocTracker.hpp"
```

(or, apply my suggestions.)

Cheers, Thomas

**Comment by RealCLanger** [comment]
> Hi Christioph,
> 
> thanks for doing this. Ignore my earlier approval, this needs some fixes to build with 16:
> 

Thanks, Thomas. I updated the change with your suggestions.

**Comment by SapMachine** [comment]
Hello @RealCLanger, this pull request fulfills all formal requirements.

**Review by tstuefe — APPROVED** [review]
All good now. Thanks!

**Review by schmelter-sap — APPROVED** [review]
LGTM

**Pull request merged**